packages/snaps-controllers/src/test-utils/controller.ts

```diff
@@ -65,6 +67,56 @@ export class MockControllerMessenger<
   }
 }
 
+export class MockApprovalController {
+  #approval?: {
+    request: Partial<ApprovalRequest<Record<string, Json>>>;
+    promise: {
+      resolve: (value?: unknown) => void;
+      reject: (value?: unknown) => void;
+    };
+  };
+
+  async addRequest(request: { requestData?: Record<string, Json> }) {
+    const promise = new Promise((resolve, reject) => {
+      this.#approval = {
+        promise: { resolve, reject },
+        request,
+      };
+    });
+
+    return promise;
+  }
+
+  updateRequestStateAndApprove({
+    requestState,
+  }: {
+    requestState: Record<string, Json>;
+  }) {
+    if (this.#approval) {
+      if (requestState.loading === false && !requestState.error) {
+        this.#approval.promise.resolve({
+          permissions: requestState.permissions,
+          ...this.#approval.request.requestData,
+        });
+      }
+    }
+  }
+
+  updateRequestStateAndReject({
+    requestState,
+  }: {
+    requestState: Record<string, Json>;
+  }) {
+    if (this.#approval) {
+      if (requestState.loading === false && !requestState.error) {
+        this.#approval.promise.reject(ethErrors.provider.userRejectedRequest());
+      }
+    }
+  }
+}
+
+export const approvalControllerMock = new MockApprovalController();
```

Comment:
Should this not be defined per messenger? 🤔

Reply:
Or is there a good reason to define it globally

Reply:
It was to avoid code duplication, it will never be used at the same time by two tests so there si real reason of creating new instances by messenger. It allows me to use it inside the test file for the cases where I need to re-define the behaviour of those.